### PR TITLE
Modify Outerloop job in groovy to run on Ubuntu and Windows.

### PR DESCRIPTION
**Pull request by Priya91**
cc @mmitche @joshfree @rahulkotecha @vijaykota

### CODE REVIEW

**Comment by Priya91** [comment]
@dotnet-bot Test OuterLoop Ubuntu Debug

File: src/System.Net.Security/tests/Scripts/setup-kdc.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+OS=`cat /etc/os-release | grep "PRETTY_NAME" | sed 's/PRETTY_NAME=//g' | sed 's/["]//g' | awk '{print $1}'`
```
Comment (joshfree) [inline src/System.Net.Security/tests/Scripts/setup-kdc.sh]: Why parse PRETTY_NAME when you can just read ID from os-release?

File: src/System.Net.Security/tests/Scripts/setup-kdc.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		if pgrep kadmind 2> /dev/null; then killall kadmind ; fi
+		
+		kdc_setup
+		
+		echo "Starting KDC.."
+		${krb5kdc}
+		
+		;;
+		
+	"Debian")
```
Comment (joshfree) [inline src/System.Net.Security/tests/Scripts/setup-kdc.sh]: why not just echo ${OS}

File: src/System.Net.Security/tests/Scripts/setup-kdc.sh (hunk 1 above)
Comment (Priya91) [inline src/System.Net.Security/tests/Scripts/setup-kdc.sh]: @rahulkotecha Can you please address the PR comments on networking scripts. Thanks!

**Comment by Priya91** [comment]
Need dotnet/dotnet-ci#164 to test the ubuntu outerloop changes.

**Comment by Priya91** [comment]
test ci please

**Comment by Priya91** [comment]
Test OuterLoop Ubuntu Debug

**Comment by mmitche** [comment]
@Priya91 Your netci changes will not affect any runs here.  This will need to be merged first.

Question: Are there implications of running multiple runs on the same machine at the same time here?

**Comment by Priya91** [comment]
@mmitche Yes we don't want to do that. But we run only one run per machine at a time right?

**Comment by Priya91** [comment]
Also it would be best if we can bring this setup step further down below to xunit setup, as these tests will surely fail on local outerloop dev box runs.

**Comment by mmitche** [comment]
In the auto-images, yes

File: netci.groovy [hunk 3]
```diff
@@ -100,12 +102,13 @@ branchList.each { branchName ->
 }
 
 // **************************
-// Define outerloop windows testing.  Run locally on each machine.
+// Define outerloop windows/ubuntu testing.  Run locally on each machine.
 // **************************
 
-def osShortName = ['Windows 10': 'win10', 'Windows 7' : 'win7', 'Windows_NT' : 'windows_nt']
+def osShortName = ['Windows 10': 'win10', 'Windows 7' : 'win7', 'Windows_NT' : 'windows_nt', 'Ubuntu' : 'ubuntu']
+def outerloopOSs = ['Windows 10', 'Windows 7', 'Windows_NT', 'Ubuntu']
```
Comment (mmitche) [inline netci.groovy]: Can you change this to Ubuntu14.04 to be more specific (since we have 15.10 now too)

**Comment by mmitche** [comment]
LGTM besides the minor comment.

**Comment by Priya91** [comment]
@mmitche Incorporated the 14.04 changes. @rahulkotecha @vijaykota Can you please check if executing the script as a test class initialization works? I have created a fixture template for it to test, or would you prefer it to be in jenkins setup ?

cc @joshfree

File: src/System.Net.Security/tests/Scripts/setup-kdc.sh (hunk 1 above)
Comment (vijaykota) [inline src/System.Net.Security/tests/Scripts/setup-kdc.sh]: Sure.. Rahul can take care of all these changes when he sends his PR with the updates required for other distros as well

**Comment by vijaykota** [comment]
> Can you please check if executing the script as a test class initialization works

We will try this out. I believe this should work since the Jenkins user now has sudo perms.

LGTM for Priya's changes. Thanks!

File: src/System.Net.Security/tests/Scripts/setup-kdc.sh (hunk 1 above)
Comment (rahulkotecha) [inline src/System.Net.Security/tests/Scripts/setup-kdc.sh]: took care of these changes and changes required for other Linux distros as part of: https://github.com/rahulkotecha/corefx/commit/e8acf953de2af1f2fce8f7e98cc6d080d6af42fb
Thanks!

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by joshfree** [comment]
The test initialization needs to be separate from Jenkins infrastructure - so that developers can execute the test outside of a Jenkins CI environment.

**Comment by Priya91** [comment]
This needs dotnet/dotnet-ci#173 before merge.

**Comment by Priya91** [comment]
The Outerloop ubuntu/OSX tests are failing because they are not being run sudo. The netci.groovy changes in the PR are not used for PR testing. These changes will be reflected after merge.

**Comment by Priya91** [comment]
@dotnet-bot test this please

**Comment by kapilash** [comment]
@Priya91
 
>Can you please check if executing the script as a test class initialization works? I have created a fixture template for it to test, or would you prefer it to be in jenkins setup ?

I tried running it as a fixture, but could not succeed. I need to execute the script as a super user, otherwise it would fail to install the required packages or setup the KDC.

  But trying to run the tests as super user via ```sudo ./corerun xunit.console.exe <dll-name>``` results in  ```coreclr_initialize failed - status: 0x8007001f```

I could be wrong but this seems to have something to do with the permissions of the assemblies that are generated using our build.
 
Is there any reason to expect that this issue would not occur on CI machines?

**Comment by Priya91** [comment]
@kapilash The script does not have to be run as sudo, since I have made the individual instructions in the script that need to be elevated as sudo. Or if the whole script has to be sudo then there can be a wrapper to run it as ```sudo setup-kdc.sh``` from the wrapper script.

Regarding the coreclr_initialize failed, that is not related to these changes. Seems like coreclr package was not found.

**Comment by Priya91** [comment]
Have pushed the wrong branch. fixing this.

**Comment by Priya91** [comment]
```
15:30:26 Mismatch : expected: <Root><!--test - - --></Root>
15:30:26  actual: <Root>
15:30:26   <!--test - - -->
15:30:26 </Root>
15:30:26 Mismatch : expected: <Root><?badpi text ? >?></Root>
15:30:26  actual: <Root>
15:30:26   <?badpi text ? >?>
15:30:26 </Root>
15:30:26 Mismatch : expected: <Root><![CDATA[test ]]]]><![CDATA[> test]]></Root>
15:30:26  actual: <Root><![CDATA[test ]]]]><![CDATA[> test]]></Root>
```

@dotnet-bot Test Innerloop Ubuntu Release Build and Test

**Comment by Priya91** [comment]
```
16:01:57 d:\j\workspace\dotnet_corefx\osx_release_bld_prtest>call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x86   && build.cmd /p:ConfigurationGroup=Release /p:OSGroup=OSX /p:SkipTests=true /p:TestNugetRuntimeId=osx.10.10-x64 
16:01:57 Installing dotnet cli...
16:02:08 Restoring BuildTools version 1.0.25-prerelease-00165...
16:07:09 Error: FindPackagesById: Microsoft.DotNet.BuildTools
16:07:09   HTTP request timed out. Exiting.
16:07:09 ERROR: Could not restore build tools correctly. See 'd:\j\workspace\dotnet_corefx\osx_release_bld_prtest\init-tools.log' for more details.
```

**Comment by Priya91** [comment]
@dotnet-bot test this please

**Comment by Priya91** [comment]
#6153

**Comment by Priya91** [comment]
@dotnet-bot Test Innerloop OSX Release Build and Test

**Comment by Priya91** [comment]
@dotnet-bot Test Innerloop Ubuntu Release Build and Test

**Comment by Priya91** [comment]
@dotnet-bot Test Innerloop Ubuntu Release Build and Test

File: netci.groovy [hunk 4]
```diff
@@ -6,6 +6,8 @@ def project = GithubProject
 
 // Globals
 
+// List of all CI OS
+def allOSs = ['Ubuntu', 'Debian8.2', 'OSX', 'FreeBSD', 'CentOS7.1', 'OpenSUSE13.2', 'Windows 10', 'Windows 7', 'Windows_NT']
```
Comment (weshaggard) [inline netci.groovy]: Where is this used? Perhaps I missed it but I don't see it used in the script anywhere.

Comment (Priya91) [inline netci.groovy]: This is not used anywhere currently, but will be used when all OSes are supported in Outerloop. Will remove the def Outerloop then. Sorry your comment didnt update on my client.. Saw this only now.

**Pull request merged**